Makefile

```diff
@@ -122,17 +126,32 @@ ifeq ($(FIRMWARE),uefi)
 	$(MAKE) -C $(T)/misc/efi-stub HV_OBJDIR=$(HV_OUT) BOARD=$(BOARD) FIRMWARE=$(FIRMWARE) SCENARIO=$(SCENARIO) EFI_OBJDIR=$(EFI_OUT) all install-debug
 endif
 
-sbl-hypervisor-install:
-	$(MAKE) -C $(T)/hypervisor HV_OBJDIR=$(HV_OUT)-sbl/apl-mrb BOARD=apl-mrb FIRMWARE=sbl SCENARIO=sdc RELEASE=$(RELEASE) install
-	$(MAKE) -C $(T)/hypervisor HV_OBJDIR=$(HV_OUT)-sbl/apl-up2 BOARD=apl-up2 FIRMWARE=sbl SCENARIO=sdc RELEASE=$(RELEASE) install
-	$(MAKE) -C $(T)/hypervisor HV_OBJDIR=$(HV_OUT)-isd BOARD=kbl-nuc-i7 FIRMWARE=uefi SCENARIO=industry RELEASE=$(RELEASE) install
-	$(MAKE) -C $(T)/misc/efi-stub HV_OBJDIR=$(HV_OUT)-isd BOARD=kbl-nuc-i7 FIRMWARE=uefi SCENARIO=industry EFI_OBJDIR=$(EFI_OUT) all install
-
-sbl-hypervisor-install-debug:
-	$(MAKE) -C $(T)/hypervisor HV_OBJDIR=$(HV_OUT)-sbl/apl-mrb BOARD=apl-mrb FIRMWARE=sbl SCENARIO=sdc RELEASE=$(RELEASE) install-debug
-	$(MAKE) -C $(T)/hypervisor HV_OBJDIR=$(HV_OUT)-sbl/apl-up2 BOARD=apl-up2 FIRMWARE=sbl SCENARIO=sdc RELEASE=$(RELEASE) install-debug
-	$(MAKE) -C $(T)/hypervisor HV_OBJDIR=$(HV_OUT)-isd BOARD=kbl-nuc-i7 FIRMWARE=uefi SCENARIO=industry RELEASE=$(RELEASE) install-debug
-	$(MAKE) -C $(T)/misc/efi-stub HV_OBJDIR=$(HV_OUT)-isd BOARD=kbl-nuc-i7 FIRMWARE=uefi SCENARIO=industry EFI_OBJDIR=$(EFI_OUT) all install-debug
+apl-mrb-sbl-sdc:
+	$(call build_acrn,sbl,apl-mrb,CONFIG_SDC,sdc)
+apl-up2-sbl-sdc:
+	$(call build_acrn,sbl,apl-up2,CONFIG_SDC,sdc)
+kbl-nuc-i7-uefi-industry:
+	$(call build_acrn,uefi,kbl-nuc-i7,CONFIG_INDUSTRY,industry)
+
+sbl-hypervisor: apl-mrb-sbl-sdc apl-up2-sbl-sdc kbl-nuc-i7-uefi-industry
+
+apl-mrb-sbl-sdc-install:
+	$(call install_acrn,sbl,apl-mrb,sdc)
+apl-up2-sbl-sdc-install:
+	$(call install_acrn,sbl,apl-up2,sdc)
+kbl-nuc-i7-uefi-industry-install:
+	$(call install_acrn,uefi,kbl-nuc-i7,industry)
+
+sbl-hypervisor-install: apl-mrb-sbl-sdc-install apl-up2-sbl-sdc-install kbl-nuc-i7-uefi-industry-install
+
```

Comment:
Please have a check if there are multiple uefi targets, does a previous efi file will be override by the later one?

Reply:
Hi Binbin,
Double checked. There is no issue here. The .efi file is created in HV_OUT which is per build target.